erpnext/manufacturing/report/discrete_manufacture_costing/discrete_manufacture_costing.py

```diff
@@ -0,0 +1,121 @@
+                JOIN `tabStock Entry Detail` as SED
+                    ON SED.parent = SE.name
+                JOIN `tabProduction Plan` as PP
+                    ON PP.name = WO.production_plan
+                JOIN `tabProduction Plan Sales Order` as SOD
+                    ON SOD.parent = WO.production_plan
+            WHERE WO.production_plan = %s AND SOD.sales_order = %s AND SE.stock_entry_type = %s
+                AND SED.item_group IN %s
+            GROUP BY SOD.sales_order, WO.name, WO.production_plan, SED.item_code """, 
+        (production_plan, sale_order, stock_entry_type, filterItemGroup), as_list=1)
+    else:
+        response = frappe.db.sql(""" SELECT SOD.sales_order, WO.name, WO.production_plan, 
+                SED.item_code, SE.stock_entry_type, SED.item_group, 
+                SUM(ROUND(SED.qty, 3)), SED.uom, SED.basic_rate,
+                (SED.qty * SED.basic_rate) as amount
+            FROM `tabWork Order` as WO
+                JOIN `tabStock Entry` as SE
+                    ON SE.work_order = WO.name
+                JOIN `tabStock Entry Detail` as SED
+                    ON SED.parent = SE.name
+                JOIN `tabProduction Plan` as PP
+                    ON PP.name = WO.production_plan
+                JOIN `tabProduction Plan Sales Order` as SOD
+                    ON SOD.parent = WO.production_plan
+            WHERE WO.production_plan = %s AND SOD.sales_order = %s AND SE.stock_entry_type = %s
+            GROUP BY SOD.sales_order, WO.name, WO.production_plan, SED.item_code """, 
+        (production_plan, sale_order, stock_entry_type), as_list=1)
+    
+    # fetch production quontity and Item rate to display summary
+    summary = frappe.db.sql(""" SELECT PP.total_produced_qty, SOD.item_code
+        FROM `tabProduction Plan` as PP
+        JOIN `tabProduction Plan Item` as SOD
+            ON SOD.parent = PP.name
+        WHERE PP.name = %s 
+        limit 1""", (production_plan), as_list=1)
+
+    # company = get_default_company()
+    # currency = get_company_currency(company)
+    # print(currency)
+
+    totalQty = 0
+    totalAmount = 0
+    totalValuation = 0
+    for sale in response:
+        totalQty = totalQty + sale[6]
+        totalAmount = totalAmount + sale[9]
+        totalValuation = totalValuation+ sale[8]
+        # sale[8] = str(sale[8]) + " " + currency
+        # sale[9] = str(sale[9]) + " " + currency
+    
+    totalCostPerUOM = 0
+    if (totalAmount != 0):
+        totalCostPerUOM = float("{0:.3f}".format(totalAmount / summary[0][0]))
+    
+    totalAmount = float("{0:.3f}".format(totalAmount))
+    totalQty = float("{0:.3f}".format(totalQty))
+    totalValuation = float("{0:.3f}".format(totalValuation))
+
+    # temporary quick solution
+    response.append(['', '', '', '<b>Total</b>', '', '', totalQty, '', totalValuation, totalAmount])
+    response.append(['', '', '', '<b>Manufacturing Cost per UOM</b>', '', '', '', '', '', totalCostPerUOM])
```

Comment:
It's always difficult to maintain proper indexing to map the data in proper column if we use a `list`.

Instead we should use a `dict`.

```
response.append({
    "stock_entry_type": _("Manufacturing Cost per UOM"),
    'amount': total_cost_per_uom
})
```